tests/unit/FirebaseObject.spec.js

```diff
@@ -322,6 +322,21 @@ describe('$FirebaseObject', function() {
       expect($scope.test).toEqual({$value: null, $id: obj.$id, $priority: obj.$priority});
     });
 
+    it('should delete $value if set to an object', function () {
```

Comment:
Is this strictly true? I understand why this is the case for `null`, but I don't see why setting it to some other object (not that you'd really want to do that), would delete `$value`. I don't think that is expected nor what we would want to happen.